src/app/base/components/PrefixedInput/PrefixedInput.tsx

```diff
@@ -9,45 +8,42 @@ export type PrefixedInputProps = Omit<InputProps, "type"> & {
   immutableText: string;
 };
 
-// TODO: Upstream to maas-react-components https://warthogs.atlassian.net/browse/MAASENG-3113
 const PrefixedInput = ({ immutableText, ...props }: PrefixedInputProps) => {
-  const prefixedInputRef: RefObject<HTMLDivElement> = useRef(null);
+  const prefixTextRef = useRef<HTMLDivElement>(null);
+  const inputWrapperRef = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
-    const inputWrapper = prefixedInputRef.current?.firstElementChild;
-    if (inputWrapper) {
-      if (props.label) {
-        // CSS variable "--immutable" is the content of the :before element, which shows the immutable octets
-        // "--top" is the `top` property of the :before element, which is adjusted if there is a label to prevent overlap
-        inputWrapper.setAttribute(
-          "style",
-          `--top: 2.5rem; --immutable: "${immutableText}"`
-        );
-      } else {
-        inputWrapper.setAttribute("style", `--immutable: "${immutableText}"`);
-      }
-
-      const width = window.getComputedStyle(inputWrapper, ":before").width;
+  useLayoutEffect(() => {
```

Comment:
`useLayoutEffect` runs synchronously immediately after React has performed all DOM mutations, but before the browser has had a chance to paint those changes. This avoids any potential visual glitches or layout shifts.